packages/video_player/video_player_avfoundation/darwin/video_player_avfoundation/Sources/video_player_avfoundation/FVPVideoPlayerPlugin.m

```diff
@@ -451,6 +451,8 @@ - (void)setupEventSinkIfReadyToPlay {
     }
 
     BOOL hasVideoTracks = [asset tracksWithMediaType:AVMediaTypeVideo].count != 0;
+    // In HLS streams, AVAsset may not always provide track information because it is a streaming format.
+    // For audio-only HLS files, track information must be checked using `currentItem.tracks.count`.
```

Comment:
This doesn't explain why for audio only HLS, track info must be checked using `currentItem.tracks`. What's special about audio only HLS?

Reply:
@hellohuanlin 
Thank you for your feedback! I improved the comment to clear the reason (due to the format and no size):
[Improve the comment for checking HLS audio-only tracks](https://github.com/flutter/packages/pull/7890/commits/fae1d41ddc74af2ef2a9a89edb75f5f99539a77a)